sam/models/base_model.py

```diff
@@ -76,7 +79,9 @@ def __init__(
         self.use_diff_of_y = use_diff_of_y
         self.timecol = timecol
         self.y_scaler = y_scaler
-        self.feature_engineer_ = feature_engineer if feature_engineer else IdentityFeatureEngineer
+        self.feature_engineer_ = (
+            feature_engineer if feature_engineer else IdentityFeatureEngineer()
+        )
 
     @abstractmethod
     def get_untrained_model(self) -> Callable:
```

Comment:
Should we not raise an NotImplemented error here?

Reply:
No, I think NotImplementedError is better, see: https://theflashreads.com/004-notimplemented-notimplementederror/